### DISCUSSION TITLE
Bad NER accuracy

### DISCUSSION THREAD

**Discussion by orglce** [explosion/spaCy#11350] [Help: Model Advice]
I am trying to train an NER model on the Universal Dependency dataset for [Slovenian](https://github.com/UniversalDependencies/UD_Slovenian-SSJ). I am using the spacy project [ner_demo](https://github.com/explosion/projects/tree/v3/pipelines/ner_demo). The project.yml and config file were left untouched and the training was done for some 50ish epochs, but the the accuracy of the predictions for NER (f score) always stayed around 20%. The results are the same when training in conjunction with the full pipeline (tagger, parser etc.).

The dataset contains NER annotations for around 90% of the data, and the left is unannotated. I have also tried training with just the annotated data and results are still the same. Could there be a problem with the dataset or am I missing some crucial configuration steps?

**Comment by adrianeboyd** [comment]
There's no official format for NER annotation in CoNLL-U, so it's sounds like it's an underlying data quality or a data conversion issue. (In addition, I think it's unusual for corpora distributed in UD repos to include it?)

`spacy convert` supports some formats used in `MISC` by existing corpora like NorNE, but it's just ad-hoc. I don't think it supports the format in this corpus, so double-check the training data. And one thing it definitely does not support is missing values, which it would probably treat as `O` (I'd have to double-check), so that could cause problems.

**Reply by orglce** [reply]
I have converted the annotations to the ones that Spacy supports. And when I convert the data to .spacy format, the annotations are there. But when I run the training the accuracy is really low. There are around 8000 annotated entities in the UD dataset. Is that not enough of data  to achieve higher accuracy?

**Reply by adrianeboyd** [reply]
That sounds like it should be enough data to get started as long as the annotation quality is reasonable.

I tried it out with similar results, so my current guess is that there are problems with the underlying data, but I can't read/review the annotation well enough to verify this. I would guess that possibly many entities aren't labeled consistently (sometimes `O`, sometimes labeled) or that the individual entity types aren't applied consistently. Maybe you can track down more information in the original corpus?
